flink-runtime/src/main/java/org/apache/flink/runtime/memory/AbstractPagedInputView.java

```diff
@@ -160,6 +160,11 @@ protected final void advance() throws IOException {
 		this.positionInSegment = this.headerLength;
 	}
 
+	/**
+	 * This method will run after advance(). Override it if you want to use it.
+	 */
+	protected void afterAdvance() {}
```

Comment:
Is afterAdvance() just to support setting initial offset of InputView? It's a little too heavy to add an method afterAdvance(). Consider override advance() in the HeaderlessChannelReaderInputView and set positionInSegment the first time?